analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionEmptinessChecking.cs

```diff
@@ -36,115 +36,70 @@ public sealed class CollectionEmptinessChecking : SonarDiagnosticAnalyzer
+                            && TryGetCountCall(expression, c.SemanticModel, out var location, out var typeArgument))
+                        {
+                            c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, location, typeArgument));
+                        }
+                    }
                 },
+                SyntaxKind.GreaterThanExpression,
+                SyntaxKind.GreaterThanOrEqualExpression,
+                SyntaxKind.LessThanExpression,
+                SyntaxKind.LessThanOrEqualExpression,
+                SyntaxKind.EqualsExpression,
                 SyntaxKind.NotEqualsExpression);
-        }
-
-        private static void CheckCountZero(ExpressionSyntax zero, ExpressionSyntax count, SyntaxNodeAnalysisContext context)
-        {
-            if (ExpressionNumericConverter.TryGetConstantIntValue(zero, out var value) &&
-                value == 0 &&
-                TryGetCountCall(count, context.SemanticModel, out var reportLocation, out var typeArgument))
-            {
-                context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, reportLocation, typeArgument));
-            }
-        }
-        private static void CheckCountOne(ExpressionSyntax one, ExpressionSyntax count, SyntaxNodeAnalysisContext context)
-        {
-            if (ExpressionNumericConverter.TryGetConstantIntValue(one, out var value) &&
-                value == 1 &&
-                TryGetCountCall(count, context.SemanticModel, out var reportLocation, out var typeArgument))
-            {
-                context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, reportLocation, typeArgument));
-            }
-        }
 
         private static bool TryGetCountCall(ExpressionSyntax expression, SemanticModel semanticModel, out Location countLocation, out string typeArgument)
         {
             countLocation = null;
             typeArgument = null;
             var invocation = expression as InvocationExpressionSyntax;
-            if (!(invocation?.Expression is MemberAccessExpressionSyntax memberAccess))
-            {
-                return false;
-            }
 
-            if (!(semanticModel.GetSymbolInfo(memberAccess).Symbol is IMethodSymbol methodSymbol) ||
-                !IsMethodCountExtension(methodSymbol) ||
-                !methodSymbol.IsExtensionOn(KnownType.System_Collections_Generic_IEnumerable_T))
+            if ((invocation?.Expression is MemberAccessExpressionSyntax memberAccess)
+                && (semanticModel.GetSymbolInfo(memberAccess).Symbol is IMethodSymbol methodSymbol)
+                && IsMethodCountExtension(methodSymbol)
+                && methodSymbol.IsExtensionOn(KnownType.System_Collections_Generic_IEnumerable_T))
             {
-                return false;
-            }
+                if (methodSymbol.IsGenericMethod)
+                {
+                    typeArgument = methodSymbol.TypeArguments.First().ToDisplayString();
```

Comment:
There should be just one.
```suggestion
                    typeArgument = methodSymbol.TypeArguments.Single().ToDisplayString();
```